lib/get-exporter.js

```diff
@@ -0,0 +1,120 @@
+const paginateRequest = require('./paginate-request');
+const path   = require('path');
+const inquirer = require('inquirer');
+const {
+  loadConfig,
+  getShallowStatus,
+  logResult,
+  logProcessing,
+  logError,
+  mapById,
+  plural
+} = require('./utils');
+const {
+  remove,
+  pathExists,
+  writeFile,
+  ensureDir
+} = require('fs-extra');
+const { merge, values, isEmpty } = require('omnibelt');
+const allSettledSerial = require('./all-settled-serial-p');
+const { rollbarLog } = require('./rollbar');
+
+const getExporter = ({
+  apiType, commandType, localStatusParams, remoteStatusParams, getData, extraQuery
+}) => {
+  const exporter = async (pattern, command = {}, loadedConfig) => {
+    let didExport = false;
+    const { apiToken, applicationId, api } = loadedConfig ? loadedConfig : await loadConfig();
+    if (!apiToken || !applicationId) { return; }
+    let items;
+    try {
+      let query = { applicationId };
+      if (extraQuery) {
+        query = merge(query, extraQuery);
+      }
+      items = await paginateRequest(api[apiType].get, query);
+    } catch (e) {
+      return logError(e);
+    }
+    const itemsById = mapById(items);
+    const { remoteStatusByFile, localStatusByFile } = await getShallowStatus({
```

Comment:
I would call this localStatusList or something else cause it's an array an not a object keyed by file name

Reply:
if you want to transform this into a keyValue pair I would change it to use a Map(), and then instead of doing stuff like `localStatusByFile.indexOf(remoteStatus.file) > 0` you can do localStatusByFile.has(remoteStatus.file)

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map